jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/BaseImageParameters.java

```diff
@@ -35,23 +35,32 @@
   private final PlatformParametersSpec platformParametersSpec;
   private final ListProperty<PlatformParameters> platforms;
 
+  private final ObjectFactory objectFactory;
+
   @Inject
   public BaseImageParameters(ObjectFactory objectFactory) {
+    this.objectFactory = objectFactory;
     auth = objectFactory.newInstance(AuthParameters.class, "from.auth");
     platforms = objectFactory.listProperty(PlatformParameters.class).empty();
     platformParametersSpec =
         objectFactory.newInstance(PlatformParametersSpec.class, objectFactory, platforms);
-
-    PlatformParameters platform = new PlatformParameters();
-    platform.setOs("linux");
-    platform.setArchitecture("amd64");
-    platforms.add(platform);
   }
 
   @Nested
   @Optional
   public ListProperty<PlatformParameters> getPlatforms() {
-    return platforms;
+    if (!platforms.get().isEmpty()) {
```

Comment:
It does feel a little weird that a `getter` is mutating things.

Reply:
could we just delegate default setting to a lower level?

Reply:
Technically it doesn't mutate an internal state, but it's true the class is no longer a POJO or bean. I also didn't like it and wondered other solutions yesterday. Setting the default at other levels could be another way, but it also had some cons.

Today, I think I found a better way than these: start the class with the default value as before, but only reset it when the user specifies `platforms {}`. This behavior is also more or less similar to how it works in Maven; if nothing is defined, Maven assumes amd64/linux, but as soon as the user specifies `<platforms>`, it becomes an empty list.. What do you think?

Reply:
Hrmm, I just realized I don't understand how this affects verifying image compatibility? Do we have code somewhere that verifies and image matches the specified platform?

Reply:
#2781 and this PR have nothing to do with verifying image platforms. The issue was that defining
```
jib.from.platforms {
  platform {
    architecture = 'non-amd64'
    os = '...'
  }
}
```
incorrectly resulted in *two* platforms configured: the configured one plus the default amd64/linux. This was because `platform { ...}` just called `ListProperty.add()`.